### DISCUSSION TITLE
Comment input box too small

### DISCUSSION THREAD

**Discussion by tacerus** [community/community#14188] [Projects and Issues]
Hi,

typing comments with code snippets tends to get tedious for me, as the box is too small to fit my content, forcing me to scroll up and down a lot, never being able to get a full glance at what I typed:

![image](https://user-images.githubusercontent.com/5121046/161435671-9c486e12-ca43-43f2-9643-5f46b3d3d044.png)

It would be nice if the box automatically enlarged itself if content grows out of bounds of the box.
Alternatively (or even additionally!), making the edge of the box "draggable" to manually stretch the size would be an improvement as well.

Is there possibly already some browser script enabling either of these?

Thanks!

**Comment by mantono** [comment]
I agree. Sometimes I feel like I am not meant to give detailed and constructive feedback on pull requests. The interface works well for comments like "LGTM" or "👍🏼" but is an annoyance for longer comments. I usually end up writing my longer comments somewhere else first, and the copying it to GitHub when I am done writing it.

**Comment by essexe** [comment]
I was Google searching for how to fix this problem when I came to this page. The text box being too small is definitely annoying. I eventually discovered it's possible to make it larger by dragging the little widget in the bottom right. It works for me, in Firefox 105.0. 
![image](https://user-images.githubusercontent.com/12091534/192042532-cb8c34c5-b9f2-44ab-9545-4292ed194a96.png)

**Reply by mantono** [reply]
That is great, **thank you**! I could still wish that the default size was slightly larger, or even better, it would increase in size (at least some to extent) when the text cannot fit the text area.

I tried your tip when writing this comment, and it works well. But for me the area to drag with is super small, I had never noticed it was there until you mentioned it now. Perhaps that is a due to my system theme or something (I am using Firefox 104).

![input](https://user-images.githubusercontent.com/7093365/192046859-0d8581ab-0c17-4f00-87d6-60edfe9180d4.png)

**Reply by tacerus** [reply]
Apparently my dark theme makes these "ridges" quite invisible as well. Now if only it would save the size.. but then again, I should not be so demanding and be grateful for this advancement! :-)

Thank you, @essexe.

**Reply by frdij** [reply]
I did find I can change the _height_ however, I would very much like to be able to change the width as well. The default width is so narrow and code seggestions often don't fit without wrapping even though the code lien isn't that wide.

So much wasted space, especially when you've got an ultra-wide monitor (3440px), but even on normal width monitor (1920px). :(

![image](https://github.com/user-attachments/assets/94a6316a-f09a-4732-acc8-5aed1de2c47e)
![image](https://github.com/user-attachments/assets/2786a7de-33fb-43ab-91c1-3aa0633f386d)

**Comment by roll** [comment]
I completely agree. I was always aware of the text box enlargement handle but it's too annoying to use it for every comment you need to write. Actually, I found this thread searching for `github settings comment box height` wondering if it's configurable

**Comment by Stickerpants** [comment]
Dragging the box to enlarge is a good workaround, but I would really love to see this be configurable - otherwise, it's a pain to have to fiddle with every single comment.

**Comment by nelsonspbr** [comment]
+1

**Comment by dannz510** [comment]
That's a very common frustration for developers and anyone working with code in text areas\! You're absolutely right that having to constantly scroll in a fixed-size comment box when dealing with code snippets is tedious and hinders your ability to review your content at a glance. The inability to see your full code, especially with the automatic wrapping, makes precise editing difficult.

The suggestions you have – **auto-enlarging the box** and **making it manually resizable** – are standard and highly desired features for any rich text editor, especially one used for code.

Here's a breakdown of your points and potential solutions:

### Product Feedback: Resizable Comment/Text Area for Code Snippets 📏

You're seeking an improvement to the comment/text editor experience within the platform (likely GitHub, given your previous questions), specifically when typing comments that include code snippets.

**Current Pain Points:**

  * **Fixed Size:** The current comment box has a fixed height, making it difficult to view and edit longer code blocks.
  * **Excessive Scrolling:** Users are forced to scroll up and down constantly to see their content, which is inefficient and disrupts workflow.
  * **Limited Overview:** It's hard to get a full overview of the code or comment, leading to potential errors or missed details.

**Proposed Solutions:**

1.  **Automatic Resizing:** The comment box should automatically expand vertically as content is added, accommodating longer text and code snippets without forcing manual scrolling.
2.  **Manual Resizing (Draggable Edge):** Implement a draggable edge (typically at the bottom right corner) that allows users to manually resize the comment box to their preferred height and width.

### Is there already a browser script enabling either of these?

This is an excellent question\! For many web applications that lack built-in resizing for text areas, browser extensions or user scripts (like those managed by Tampermonkey or Greasemonkey) can sometimes provide this functionality.

  * **Browser Extensions:** You might find browser extensions specifically designed to make text areas on *all* websites resizable or auto-growing. Search your browser's extension store (Chrome Web Store, Firefox Add-ons, etc.) for terms like "textarea resizer," "auto-growing textarea," or "github textarea resize."
      * **Caveat:** These are generic solutions and might not always work perfectly with every website's unique CSS and JavaScript, or they might cause unexpected layout issues.
  * **User Scripts (Tampermonkey/Greasemonkey):** If no direct extension works well, a custom user script could be an option. This would involve writing a small piece of JavaScript that specifically targets the comment box element on GitHub (or whatever platform you're using) and applies resizing logic.
      * **Requires Coding Knowledge:** This option requires some basic JavaScript knowledge to create or adapt existing scripts. You'd need to identify the CSS selector for the comment `textarea` element and then apply styles or event listeners to enable resizing.
      * **Community Scripts:** Check user script repositories (like Greasy Fork) for existing scripts related to GitHub that might offer this functionality.

**Example of a conceptual user script (not a full solution, but to illustrate the idea):**

```javascript
// ==UserScript==
// @name         GitHub Comment Textarea Resizer
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  Makes GitHub comment textareas resizable
// @author       Your Name
// @match        https://github.com/*
// @grant        none
// ==/UserScript==

(function() {
    'use strict';

    // Function to apply resizing
    function makeResizable() {
        // Target the comment textareas (adjust selector as needed)
        const textareas = document.querySelectorAll('textarea[aria-label="Leave a comment"]'); // Example selector, verify on GitHub
        textareas.forEach(textarea => {
            if (!textarea.style.resize) { // Apply only if not already set
                textarea.style.resize = 'vertical'; // Allow vertical resizing
                textarea.style.overflow = 'auto'; // Ensure scrollbars appear if content overflows before manual resize
                // For auto-grow, you'd add an input event listener
                // textarea.addEventListener('input', function() {
                //    this.style.height = 'auto';
                //    this.style.height = (this.scrollHeight) + 'px';
                // });
            }
        });
    }

    // Run when the page loads, and potentially when new content is loaded dynamically
    makeResizable();

    // Use a MutationObserver to detect when new textareas might be added to the DOM (e.g., when opening a new comment box)
    const observer = new MutationObserver(makeResizable);
    observer.observe(document.body, { childList: true, subtree: true });

})();
```

**Recommendation:**

Your feedback is very valuable and aligns with standard UI/UX best practices for text editors. You should definitely **submit this as a direct feature request to GitHub.** For immediate relief, explore browser extensions, and if you're comfortable with a bit of code, a user script could be a custom solution.

**Comment by pycckuu** [comment]
+1

**Comment by jankratochvil** [comment]
It works automatically in Chromium but in Firefox one needs to resize it manually as suggested above.

**Reply by frdij** [reply]
That is only for the height. There should also be an option to increase the width.
